scrapy/pipelines/files.py

```diff
@@ -334,6 +334,9 @@ def _warn():
 
         media_guid = hashlib.sha1(to_bytes(url)).hexdigest()  # change to request.url after deprecation
         media_ext = os.path.splitext(url)[1]  # change to request.url after deprecation
+        if not media_ext or not all([char.isalpha() or char.isdigit() for char in media_ext[1:]]):
+            media_base_url = url.split('?', 1)[0]  # change to request.url after deprecation
```

Comment:
`file_path` method receives a `Request` as argument so instead of using split; you can use [`urlparse_cached`](https://github.com/scrapy/scrapy/blob/master/scrapy/utils/httpobj.py#L8) and use `.path`.